### PR TITLE
Refactoring two source folders

### PR DESCRIPTION
**Pull request by ChenPeleg**
Did a big change that does not break the app.
Suppose to work at Builds and in dev. 

Documented most of the problematics parts. 

Added sanity checks To the html and Files.

### CODE REVIEW

File: shared/Reusble.jsx [hunk 1]
```diff
@@ -39,20 +38,34 @@ export const Button = ({
   extraClass,
   children,
   black,
-
+  isCommingSoon,
```
Comment (alonisser) [inline shared/Reusble.jsx:41]: typo: should be isComingSoon

File: shared/Reusble.jsx [hunk 2]
```diff
@@ -606,7 +614,7 @@ export function TrashButton({ handleDelete, fieldValues, index }) {
   return (
     <button
       type="button"
-      aria-label="delete this field"
+      // disabled={!fieldValues[index].id}
```
Comment (alonisser) [inline shared/Reusble.jsx:617]: wat? how is this connected?

**Review by alonisser — DISMISSED** [review]

File: package.json [hunk 3]
```diff
@@ -4,8 +4,14 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
-    "dev": "vite",
+    "dev": "vite ",
+    "dev:contrast": "vite --mode contrast_prod ",
+    "dev:uncontrast": "vite --mode uncontrast_prod ./index-uncontrast.html",
     "build": "vite build",
```
[the remark this answers is not in the document]
Comment (ChenPeleg) [reply package.json:10]: Yea, no changes to the current situation

File: package.json [hunk 4]
```diff
@@ -4,8 +4,14 @@
...
+    "build:contrast": "vite build --mode contrast_prod",
+    "postbuild:contrast": "node -e \"require('fs').rename('dist-contrast/index-contrast.html', 'dist-contrast/index.html', ()=>{})\"",
```
[the remark this answers is not in the document]
Comment (ChenPeleg) [reply package.json:12]: Yes... That is the reason I did it like that...

File: index-uncontrast.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <title>UnConTraSt</title>
+    <!-- <script src="negeshim.js" type="text/javascript"></script>
+    <script type="text/javascript">
+      accessibility_rtl = false;
+      pixel_from_side = -5;
+      pixel_from_start = 200;
+  </script> -->
+      <script type="module">
+          //This is needed for dev server to run see documentation/dev-server-plugin-fix
+          import RefreshRuntime from "/@react-refresh"
```
[the remark this answers is not in the document]
Comment (ChenPeleg) [reply index-uncontrast.html:21]: Good idea. I'll try

File: shared/Reusble.jsx (hunk 2 above)
Comment (ChenPeleg) [reply shared/Reusble.jsx:617]: @oryrosin I'm adding a TODO for you to check it out latetr

**Comment by ChenPeleg** [comment]
image
What is this ? I'm merging into main

File: vite.config.js [hunk 6]
```diff
@@ -1,9 +1,72 @@
...
-export default defineConfig({
-  plugins: [react(), svgr(), visualizer()],
+export default defineConfig(({command, mode, isSsrBuild, isPreview}) => {
+    const defaultConfig = {
+        plugins: [react(), svgr(), {
+            name: 'index-html-build-replacement', apply: 'serve', //   docs:
+            //   https://vitejs.dev/guide/api-plugin.html#conditional-application
+            async transformIndexHtml(html) {
+                const tags = [];
+                if (command === 'serve') {
```
Comment (ChenPeleg) [inline vite.config.js:15]: @alonisser This is the most elegant way I could find of injecting the script only on dev environment.

**Review by ChenPeleg — COMMENTED** [review]
Injecting the react update script only on dev

File: shared/Reusble.jsx (hunk 2 above)
Comment (oryrosin) [reply shared/Reusble.jsx:619]: i dont see it on my machine... same branch on starting point

**Comment by ChenPeleg** [comment]
This has a lot of non relevant changes. Closing

**Pull request closed without merging**